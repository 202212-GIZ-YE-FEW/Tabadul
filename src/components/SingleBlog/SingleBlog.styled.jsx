import styled from "@emotion/styled";

export const BlogContainer = styled.div`
    margin-left: auto;
    margin-right: auto;
    padding-left: 15px;
    padding-right: 15px;
    @media (min-width: 768px) {
        width: 750px;
    }
    /* Medium */
    @media (min-width: 992px) {
        width: 970px;
    }
    @media (min-width: 1200px) {
        width: 1170px;
    }
    /* Large */
    @media (min-width: 1500px) {
        width: 1400px;
    }
`;
export const Blog = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 132px;
`;
export const ContentSide = styled.div`
    text-align: left;
    flex: 1;
`;
export const ImageSide = styled.div`
    width: 100%;
    height: 100%;
    flex: 1;
    text-align: center;
`;
export const BlogHeading = styled.h2`
    font-style: normal;
    font-weight: 700;
    font-size: 32px;
    line-height: 48px;
    color: #000000;
`;
export const BlogParagraph = styled.p`
    font-style: normal;
    font-weight: 400;
    font-size: 24px;
    line-height: 36px;
    color: #000000;
    width: 657px;
`;
export const BlogButton = styled.button`
    color: #fff;
    background-color: #33956d;
    border-radius: 82px;
    font-style: normal;
    font-weight: 500;
    font-size: 24px;
    line-height: 36px;
    text-align: center;
    width: 205px;
    height: 49.16px;
    border: none;
`;
