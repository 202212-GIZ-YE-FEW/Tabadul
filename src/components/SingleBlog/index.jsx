import Link from "next/link";
import React from "react";

import {
    Blog,
    BlogButton,
    BlogHeading,
    BlogImge,
    BlogParagraph,
    ContentSide,
    ImageSide,
} from "./SingleBlog.styled";
import photo from "../../assets/Image/Photo.svg";

function SingleBlog({ title, desc, id }) {
    return (
        <Blog>
            <ContentSide>
                <BlogHeading>{title}</BlogHeading>
                <BlogParagraph>{desc}</BlogParagraph>

                <BlogButton>
                    <Link href={`Blogs/${id}`}>
                        <span>Read More</span>
                    </Link>
                </BlogButton>
            </ContentSide>
            <ImageSide>
                <BlogImge src={photo} alt='BlogPhoto' priority />
            </ImageSide>
        </Blog>
    );
}

export default SingleBlog;
