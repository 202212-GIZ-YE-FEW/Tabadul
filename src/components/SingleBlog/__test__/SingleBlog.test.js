import renderer from "react-test-renderer";

import SingleBlog from "..";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

it("renders correctly", () => {
    const tree = renderer.create(<SingleBlog />).toJSON();
    expect(tree).toMatchSnapshot();
});
