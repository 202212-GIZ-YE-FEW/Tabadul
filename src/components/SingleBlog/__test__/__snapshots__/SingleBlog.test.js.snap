// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-10ofioq"
  data-aos="fade-up"
  data-aos-duration="1000"
>
  <div
    className="css-4nncao"
  >
    <h2
      className="css-1gu5a0a"
    />
    <p
      className="css-2496h4"
    >
      ...
    </p>
    <button
      className="css-l0swh0"
    >
      <a
        href="Blogs/undefined"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <span>
          ReadMore
        </span>
      </a>
    </button>
  </div>
  <div
    className="css-10lyku1"
  >
    <img
      alt="BlogPhoto"
      className="css-iuffjg"
      data-nimg="1"
      decoding="async"
      height={432}
      onError={[Function]}
      onLoad={[Function]}
      src=""
      style={
        Object {
          "color": "transparent",
        }
      }
      width={661}
    />
  </div>
</div>
`;
